dialogy/types/entity/time/__init__.py

```diff
@@ -189,7 +192,21 @@ def set_entity_type(self) -> None:
             self.entity_type = const.TIME
 
     @classmethod
-    def from_duckling(cls, d: Dict[str, Any], alternative_index: int) -> TimeEntity:
+    def pick_constrained_duckling_entities(cls, d: Dict[str, Any], constraints: Dict[str, Any]) -> Dict[str, Any]:        
+        grain = d[const.VALUE][const.VALUES][0][const.GRAIN]
+        if grain in const.TIME_UNITS and (constraint := constraints.get(const.TIME)):
```

Comment:
Let's exit on the converse condition instead. It will help reducing indents as well. 
I try to name my conditions for readability.

```python
is_time = grain in const.TIME_UNITS
constraints = constraints.get(const.TIME)

if not is_time or not constraints:
    return 
```
Here I realize, the function name or annotations don't tell me what commodity is expected as the output.
I think this function should take the `values` and return one of the `dicts` within.